### PR TITLE
Write Tests

### PR DESCRIPTION
**Pull request by kelvinabrokwa**

### CODE REVIEW

File: test/events.test.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+map.on('load', () => {
+
+  test('Events test', t => {
+
+    var Draw = GLDraw();
+    map.addControl(Draw);
+
+    t.test('draw.set event', t => {
+      map.once('draw.set', () => {
+        t.pass('draw.set fired on add');
```
Comment (mcwhittemore) [inline test/events.test.js]: Lets check the values passed to this events

File: test/events.test.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        t.end();
+      });
+      Draw.add(feature);
+    });
+
+
+    t.test('draw.delete event', t => {
+      map.once('draw.delete', () => {
```
Comment (mcwhittemore) [inline test/events.test.js]: ditto

File: test/events.test.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    t.test('draw.select.start event', t => {
+      map.once('draw.select.start', () => {
+        t.pass('draw.select.start fired on select');
+        t.end();
+      });
+      Draw._select(Draw.add(feature));
+    });
+
+    t.test('draw.select.end event', t => {
```
Comment (mcwhittemore) [inline test/events.test.js]: Lets get a test for when a bunch of things are deselected at once

File: test/events.test.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    t.test('draw.delete event', t => {
+      map.once('draw.delete', () => {
+        t.pass('draw.delete fired on delete');
+        t.end();
+      });
+      Draw._select(Draw.add(feature));
+      Draw._destroy();
+    });
+
+    t.test('draw.select.start event', t => {
```
Comment (mcwhittemore) [inline test/events.test.js]: Lets get a test when a bunch of things are selected at once

File: test/utils.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import mapboxgl from 'mapbox-gl';
 
-export const accessToken = 'pk.eyJ1IjoibWFwYm94IiwiYSI6IlhHVkZmaW8ifQ.hAMX5hSW-QnTeRCMAy9A8Q';
+export const accessToken = 'pk.eyJ1IjoiamZpcmUiLCJhIjoiZTFlNmQ3N2MzYmM2YzVjMzhkOTM2NTRhYzNiNGZiNGYifQ.1W47kmoEUpTJa3YIFefxUQ';
```
Comment (mcwhittemore) [inline test/utils.js]: Lets not have keys in the code. We can use env vars to help the tests run in circle if we need too.

**Comment by mcwhittemore** [comment]
@kelvinabrokwa this is looking great! Do you have a list of tests that still need to be written? I have some I'd like to make sure are covered.

**Comment by kelvinabrokwa** [comment]
@mcwhittemore not specifically, I'm eyeballing as I go. I would appreciate your input though. Feel free to edit the PR notes with a checklist.

**Comment by kelvinabrokwa** [comment]
@mcwhittemore the fix for https://github.com/mapbox/mapbox-gl-draw/issues/169 is in here. can I merge now and then continue writing tests afterwards?

I'm having trouble with floating point precision with projecting between coordinate space and lat/lons.

File: src/geometries/polygon.js [hunk 6]
```diff
@@ -38,17 +38,17 @@ export default class Polygon extends Geometry {
     if (typeof this.vertexIdx === 'undefined') {
```
Comment (mcwhittemore) [inline src/geometries/polygon.js]: @kelvinabrokwa - I don't understand `vertexIdx`. Why do we have this?

**Comment by mcwhittemore** [comment]
@kelvinabrokwa - sure! Lots of PRs for test-quest is props wise.

File: src/geometries/polygon.js (hunk 6 above)
Comment (kelvinabrokwa) [inline src/geometries/polygon.js]: it keeps track of the vertex currently being drawn

**Pull request merged**